### DISCUSSION TITLE
qwerty-keyboard as a midi controller

### DISCUSSION THREAD

**Discussion by jdfk46226** [hydrogen-music/hydrogen#1267] [Hydrogen on Linux]
Is it possible to use hydrogen with a qwerty-keyboard as the midi controller where you can assign a letter a unique sound? I am only asking because I am a student and I can't afford to buy or use a piano keyboard without it being stolen. I may have missed something in the documentation so any guidance would be appreciated.

I don't know if I am allowed to ask this but if there isn't, could you direct me to a software where it has similar functionality that does? I am a massive user of your software, and having this functionality would be invaluable to me.

**Comment by jdfk46226** [comment]
I understand that you can use the qwerty-keyboard but is there any way to reassign the letters for better convenience.

**Reply by cme** [reply]
There is not currently, but we intend to make all keyboard bindings assignable in Hydrogen 1.2.

However, it may be useful to know that the key which corresponds to a given instrument depends on the position of the instrument in the instrument list, and these can be arbitrarily changed by reordering the instruments.

http://hydrogen-music.org/documentation/manual/manual_en.html#sect.pattern_editor

**Comment by theGreatWhiteShark** [comment]
If you are using Linux, you also have a number of [applications](https://wiki.linuxaudio.org/apps/categories/virtual_midi) at hand to make your computer keyboard act as a MIDI keyboard.

**Reply by jdfk46226** [reply]
Can you advise on one that is most widely used and easiest to set up for this purpose? If I were to use VMPK, would I need to install soundfonts or is there a better alternative. Would it confuse my keyboard if I were to use it for both using hydrogen and as a midi keyboard? Thanks. Is jack-keyboard any good?

**Reply by theGreatWhiteShark** [reply]
I'm actually not an user of a virtual keyboard. But it just so happened this morning that there was an announcement for a new version of VMPK on the LAA mailing list https://lists.linuxaudio.org/archives/linux-audio-announce/2021-May/002944.html . So, this feels like a good choice since it's still in active development/maintained.

 > If I were to use VMPK, would I need to install soundfonts or is there a better alternative.

Not if you use it in combination with Hydrogen. The sounds will be generated internally based on the loaded drumkit and you only trigger them as you would using notes in patterns. (Or use it to record notes into pattern respectively).
